apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.ts

```diff
@@ -122,92 +82,101 @@ router.get(
-        logger.warn(
-          `Found a legacy job sequence (id=${fileEdit.jobSequenceId}) ` +
-            `in a file edit (id=${fileEdit.editID})`,
+    debug('Read from db');
+    const draftEdit = await readDraftEdit({
+      user_id: res.locals.user.user_id,
+      authn_user_id: res.locals.authn_user.user_id,
+      course_id: res.locals.course.id,
+      dir_name: path.dirname(relPath),
+      file_name: editorData.fileName,
+    });
+
+    if (draftEdit != null) {
+      if (draftEdit.fileEdit.job_sequence_id != null) {
+        debug('Read job sequence');
+        draftEdit.jobSequence = await getJobSequence(
+          draftEdit.fileEdit.job_sequence_id,
+          res.locals.course.id,
         );
-      } else {
-        const job = fileEdit.jobSequence.jobs[0];
-
-        debug('Found a job sequence');
-        debug(` saveAttempted=${job.data.saveAttempted}`);
-        debug(` saveSucceeded=${job.data.saveSucceeded}`);
-        debug(` syncAttempted=${job.data.syncAttempted}`);
-        debug(` syncSucceeded=${job.data.syncSucceeded}`);
-
-        // We check for the presence of a `saveSucceeded` key to know if
-        // the edit was saved (i.e., written to disk in the case of no git,
-        // or written to disk and then pushed in the case of git). If this
-        // key exists, its value will be true.
-        if (job.data.saveSucceeded) {
-          fileEdit.didSave = true;
-
-          // We check for the presence of a `syncSucceeded` key to know
-          // if the sync was successful. If this key exists, its value will
-          // be true. Note that the cause of sync failure could be a file
-          // other than the one being edited.
-          //
-          // By "the sync" we mean "the sync after a successfully saved
-          // edit." Remember that, if using git, we pull before we push.
-          // So, if we error on save, then we still try to sync whatever
-          // was pulled from the remote repository, even though changes
-          // made by the edit will have been discarded. We ignore this
-          // in the UI for now.
-          if (job.data.syncSucceeded) {
-            fileEdit.didSync = true;
+      }
+
+      if (draftEdit.jobSequence) {
+        if (draftEdit.jobSequence?.status === 'Running') {
+          // Because of the redirect, if the job sequence ends up failing to save,
+          // then the corresponding draft will be lost (all drafts are soft-deleted
+          // from the database on readDraftEdit).
+          debug('Job sequence is still running - redirect to status page');
+          res.redirect(`${res.locals.urlPrefix}/jobSequence/${draftEdit.jobSequence.id}`);
+          return;
+        }
+
+        // No draft is older than 24 hours, so it is safe to assume that no
+        // job sequence is legacy... but, just in case, we will check and log
+        // a warning if we find one. We will treat the corresponding draft as
+        // if it was neither saved nor synced.
```

Comment:
Consider removing this, it shouldn't be necessary anymore.

Reply:
Done.